### ISSUE TITLE
Warn if Docker is likely to encounter swap issues (performance impact)

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#8892]
we correcelty and nicely warn the user that their docker desktop is low on memory but still when we are Erroring, we spit out ugly messages... I suspect we could add a regex to this and make sure if this ever happens, hint to the user what is going on.

```
medya@~/workspace/minikube (crio_17_3) $ ./out/minikube start --driver=docker --base-image=local/kicbase:v0.0.10-snapshot
😄  minikube v1.12.1 on Darwin 10.15.6
✨  Using the docker driver based on user configuration
❗  Requested memory allocation (1739MB) is less than the recommended minimum 2000MB. Kubernetes may crash unexpectedly.
❗  Increase Docker for Desktop memory to at least 2.5GB or more:

        Docker for Desktop > Settings > Resources > Memory


👍  Starting control plane node minikube in cluster minikube
🔥  Creating docker container (CPUs=2, Memory=1739MB) ...
✋  Stopping "minikube" in docker ...
🔥  Deleting "minikube" in docker ...
🤦  StartHost failed, but will try again: creating host: create: creating: create kic node: container name "minikube": log: 2020-07-30T17:16:31.009963700Z  546 535 0:39 /docker/af9f0def1ca7ccd9b72c4da5e42395e3f6ab4ecb136893e606b5e1088a02bbba /sys/fs/cgroup/hugetlb rw,nosuid,nodev,noexec,relatime shared:148 master:28 - cgroup hugetlb rw,hugetlb
2020-07-30T17:16:31.010339600Z  547 535 0:40 /docker/af9f0def1ca7ccd9b72c4da5e42395e3f6ab4ecb136893e606b5e1088a02bbba /sys/fs/cgroup/pids rw,nosuid,nodev,noexec,relatime shared:149 master:29 - cgroup pids rw,pids
2020-07-30T17:16:31.010378100Z  549 535 0:42 /docker/af9f0def1ca7ccd9b72c4da5e42395e3f6ab4ecb136893e606b5e1088a02bbba /sys/fs/cgroup/systemd rw,nosuid,nodev,noexec,relatime shared:151 master:31 - cgroup cgroup rw,name=systemd'
2020-07-30T17:16:31.012576500Z  + docker_cgroup=/docker/af9f0def1ca7ccd9b72c4da5e42395e3f6ab4ecb136893e606b5e1088a02bbba: container exited unexpectedly
```

**Comment by afbjorklund** [comment]
Seems like we invest more in docker desktop support than on kubernetes recently

**Comment by briandealwis** [comment]
It would also be good to use the appropriate mappings per platform.  On macOS it should be

    Docker > Preferences > Resources > Memory

**Comment by briandealwis** [comment]
Two other things I noticed:

- A similar strings has two spaces between _Docker Desktop_ and _Settings_:
https://github.com/kubernetes/minikube/blob/e6d094009aeda27c7047b2ffa6e2ca43e5ef1018/translations/strings.txt#L632
- A similar string uses _Docker icon_:
https://github.com/kubernetes/minikube/blob/e6d094009aeda27c7047b2ffa6e2ca43e5ef1018/translations/strings.txt#L624

**Comment by briandealwis** [comment]
A futher note: my memory settings were 2GB, resulting in the following warning:
```
❗  Increase Docker for Desktop memory to at least 2.5GB or more:
			
	Docker for Desktop > Settings > Resources > Memory


❗  Requested memory allocation (1990MB) is less than the recommended minimum 2000MB. Kubernetes may crash unexpectedly.
❗  Your system has 16384MB memory but Docker has only 1990MB. For a better performance increase to at least 3GB.

	Docker for Desktop  > Settings > Resources > Memory


```

These warnings use two different recommended values, and I dutifully used the first value (2.5GB).

**Comment by tstromberg** [comment]
Fixed at head by #9126:

```

⛔  Exiting due to RSRC_DOCKER_MEMORY: Docker Desktop only has 984MiB available, less than the required 1024MiB for Kubernetes
💡  Suggestion: 

    1. Click on "Docker for Desktop" menu icon
    2. Click "Preferences"
    3. Click "Resources"
    4. Increase "Memory" slider bar to 2.25 GB or higher
    5. Click "Apply & Restart"
📘  Documentation: https://docs.docker.com/docker-for-mac/#resources
```
